### ISSUE TITLE
Unable to find Autoscaling Scheduled Action

### ISSUE THREAD

**Issue by jangrewe** [hashicorp/terraform#5900]
I have an ASG with two scheduled actions: shut it down in the evening, bring it up in the morning.
Now if i need to destroy and recreate that ASG, i'm always having the issue that it tells me this:

```
Error refreshing state: 2 error(s) occurred:

* aws_autoscaling_schedule.infra-down: Unable to find Autoscaling Scheduled Action: []*autoscaling.ScheduledUpdateGroupAction(nil)
* aws_autoscaling_schedule.infra-up: Unable to find Autoscaling Scheduled Action: []*autoscaling.ScheduledUpdateGroupAction(nil)
```

If i create any random two Scheduled Action with these names for that ASG and apply my TF (v0.6.14, btw) plan again, everything's fine! Until the next destroy of that ASG...

They are defined like this:
```
resource "aws_autoscaling_schedule" "infra-down" {
    autoscaling_group_name = "${aws_autoscaling_group.infra.name}"
    scheduled_action_name = "infra-down"
    min_size = 0
    max_size = 0
    desired_capacity = 0
    recurrence = "${var.schedule.down}"
    count = "${var.volatile}"             # only create this resource if ${var.volatile} is >0 (read: NOT in live)
}

resource "aws_autoscaling_schedule" "infra-up" {
    autoscaling_group_name = "${aws_autoscaling_group.infra.name}"
    scheduled_action_name = "infra-up"
    min_size = 0
    max_size = "${var.scaling_size.infra-max}"
    desired_capacity = "${var.scaling_size.infra-desired}"
    recurrence = "${var.schedule.up}"
    count = "${var.volatile}"             # only create this resource if ${var.volatile} is >0 (read: NOT in live)
}
```

I'm pretty sure it worked fine in some older version of TF, but unfortunately i can't tell you any specific version. =(

**Comment by vincer** [comment]
I've run into a similar issue where the scheduled actions disappear after a previous apply that resulted in the recreation of the ASG. That apply doesn't also recreate the scheduled actions, it doesn't touch them at all, so then the state thinks the old ones still exist.

**Comment by miguel-afonso-crowdmix** [comment]
I noticed that even on the same ASG if the scheduled actions go away because they were deleted manually or from a previous failed terraform run it will fail. 
During the refresh phase if state says a scheduled action exists but it can be found, refresh does not update its state and will try to delete them every time.

**Comment by leigh507** [comment]
Agreed - I have hit this issue just now. Manually deleted an ASG with some attached scaling policies and now terraform plan fails with:

`Error refreshing state: 2 error(s) occurred:

* aws_autoscaling_schedule.scale_up: Error retrieving Autoscaling Scheduled Actions: ValidationError: Group ***-asg not found
	status code: 400, request id: 8715d08a-387d-11e6-8c48-d18786d99e53
* aws_autoscaling_schedule.scale_down: Error retrieving Autoscaling Scheduled Actions: ValidationError: Group ***-asg not found
	status code: 400, request id: 8716bab8-387d-11e6-8b0a-69d7cbe498e2`

**Comment by tenyo** [comment]
Just ran into the same issue with Terraform v0.6.16

**Comment by ajlanghorn** [comment]
Have just run in to this now, on v0.6.16. I think state file manipulation is my only workaround.

**Comment by hoalix** [comment]
I get the same issue on v0.7.0. Manually modifying the state file worked as workaround.

**Comment by stack72** [comment]
Hi all

this has been fixed in #8430 - it will be released as part of 0.7.2

Paul
